### PR TITLE
chore(tests): Add HTTP -> Pipelines (no grok) -> Blackhole soak

### PR DESCRIPTION
**Pull request by leebenson**
The same as the current `http_pipelines_blackhole` soak, but replaces the translated remap calls to `true`, to assess the impact of pipelines + filtering with the exclusion of `parse_groks`.

### CODE REVIEW

**Comment by leebenson** [comment]
Note: `http_pipelines_no_grok_blackhole` should be compared against `http_pipelines_blackhole` and _not_ the baseline, since this is a new soak.

**Review by blt — APPROVED** [review]

**Comment by jszwedko** [comment]
I can see the value of this experiment, but are we sure we want this as a new soak test that runs on all PRs?

**Comment by blt** [comment]
> I can see the value of this experiment, but are we sure we want this as a new soak test that runs on all PRs?

Personally I'm happy to see a whole mess of soak tests, especially if they help while folks are working. I believe we'll eventually get to a point where culling is necessary -- consider that the ack'ing soaks will become equivalent without changes once acks become default -- but I'd rather err on the side of having too much than too little. 

That said, I think this PR and its kin suggest having some relationship mechanism between soaks would be a good idea. Some way to say "compare these tests together, in addition to the global comparison". Open to thoughts there, since it's still a vague notion for me.

**Comment by jszwedko** [comment]
> > I can see the value of this experiment, but are we sure we want this as a new soak test that runs on all PRs?
> 
> Personally I'm happy to see a whole mess of soak tests, especially if they help while folks are working. I believe we'll eventually get to a point where culling is necessary -- consider that the ack'ing soaks will become equivalent without changes once acks become default -- but I'd rather err on the side of having too much than too little.

Yeah, that's fair. I was thinking we'd eventually be constrained on soak CI resources.

> That said, I think this PR and its kin suggest having some relationship mechanism between soaks would be a good idea. Some way to say "compare these tests together, in addition to the global comparison". Open to thoughts there, since it's still a vague notion for me.

Yeah, agreed. Bruce's tests with/without acks are another good example of this.

**Comment by blt** [comment]
> > > I can see the value of this experiment, but are we sure we want this as a new soak test that runs on all PRs?
> > 
> > 
> > Personally I'm happy to see a whole mess of soak tests, especially if they help while folks are working. I believe we'll eventually get to a point where culling is necessary -- consider that the ack'ing soaks will become equivalent without changes once acks become default -- but I'd rather err on the side of having too much than too little.
> 
> Yeah, that's fair. I was thinking we'd eventually be constrained on soak CI resources.

For what it's worth we are already constrained today. I'm angling Q1 to get the soak machines autoscaling to relieve that pressure some.

**Comment by leebenson** [comment]
I expect these tests should be relatively short-lived. Once performance is at an acceptable level and bottlenecks have been resolved, ideally we'll wind up with a single test (possibly `http_pipelines_blackhole`) that rolls up the improvements.

The surface area that has the potential to impact numbers is a little sprawling currently, due to some of the work planned/in-flight. I'd like to keep these around as improvements to VRL, concurrency, etc, land. I'm also still baffled by the 0 minimum (noting the comment @blt made in Slack). I'm hoping isolating tests will reveal some insights.

TL;DR - it's a bit messy right now, but I'll clean up when I'm done 🙂
